File: Java/String-2/catDog.java
/*
Return true if the string "cat" and "dog" appear the same number of times in the given string.

catDog("catdog") ? true
catDog("catcat") ? false
catDog("1cat1cadodog") ? true
*/

public boolean catDog(String str) {
  int i,count_dog=0,count_cat=0;
  for(i=0;i<str.length()-2;i++){
    if((str.substring(i,i+3)).equals("cat"))
      count_cat++;
    if((str.substring(i,i+3)).equals("dog"))
      count_dog++;
  }
  return count_dog==count_cat;
}

File: Python/String-2/cat_dog.py
"""
Return True if the string "cat" and "dog" appear the same number of times in the given string.

cat_dog('catdog') → True
cat_dog('catcat') → False
cat_dog('1cat1cadodog') → True

"""

def cat_dog(str):
  count_cat=0
  count_dog=0
  for i in range(len(str)):
    if str[i:i+3]=="cat":
      count_cat=count_cat+1
  for i in range(len(str)):
    if str[i:i+3]=="dog":
      count_dog=count_dog+1
  return count_cat==count_dog